src/components/RestrictFocusToChildren.js

```diff
@@ -0,0 +1,116 @@
+const React = require('react');
+const _last = require('lodash/last');
+const {
+  getFocusableNodesInElement,
+  reconcileNodeArrays,
+  setNodeAttributes
+} = require('../utils/FocusManagement');
+
+const handleFocusIn = e => {
+  const lastInstance = _last(instances);
+  const nodeBeingFocused = e.target;
+  const nodeIsNotHandledInInstance = lastInstance.focusableNodesInWrapper.indexOf(nodeBeingFocused) === -1;
+
+  /**
+   * Nodes that are added to the DOM after an instance does
+   * its thing have to be handled here in the focusIn event
+   * handler.  If the node about to be focused isn't in the
+   * last instance's focusable nodes array then we redirect
+   * focus back to where it should be.
+   */
+  if (nodeIsNotHandledInInstance) {
+    const firstFocusableNodeInLastInstance = lastInstance.focusableNodesInWrapper[0];
+    const lastFocusableNodeInLastInstance = _last(lastInstance.focusableNodesInWrapper);
+
+    /**
+     * The direction the user is tabbing matters so we have to check
+     * which element was focused last and compare against the first
+     * and last elements in the instances focusable nodes array.
+     */
+
+    // Tabbing in reverse (shift+tab)
+    if (firstFocusableNodeInLastInstance === document.activeElement) {
+      lastFocusableNodeInLastInstance.focus();
+    // Tabbing forward (tab)
+    } else {
+      firstFocusableNodeInLastInstance.focus();
+    }
+  }
+};
+
+let instances = [];
+
+module.exports = class RestrictFocusToChildren extends React.Component {
+  constructor (props, context) {
+    super(props, context);
+
+    if (instances.length === 0) {
+      window.addEventListener('focusin', handleFocusIn);
+    }
+
+    instances.push(this);
+
+    this.focusableNodesInParent = [];
+    this.focusableNodesInWrapper = [];
+    this.preservedTabIndexValues = [];
+    this.previousFocusedNode = document.activeElement;
+  }
+
+  componentDidMount () {
+    this.focusableNodesInWrapper = getFocusableNodesInElement(this._wrapper);
+    this.focusableNodesInParent = reconcileNodeArrays(
```

Comment:
This isn't needed anymore since you added the `focusin` event handler, right?

Reply:
I waffled back and forth on weather or not to leave this now that we have the focusin listener.  Now that I've thought about it over night I think I will remove it as it feels redundant to aria hide and change tab indexes now.